runtime/integration-tests/src/generic/cases/oracles.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+//! Testing setups around our oracles
+
+mod ratio_provider {
+	use cfg_traits::ValueProvider;
+	use cfg_types::{
+		fixed_point::Ratio,
+		tokens::{CrossChainTransferability, CurrencyId, CustomMetadata, LocalAssetId},
+	};
+	use frame_support::traits::OriginTrait;
+	use runtime_common::oracle::Feeder;
+	use sp_runtime::traits::One;
+
+	use crate::generic::{
+		config::Runtime,
+		env::Env,
+		envs::runtime_env::RuntimeEnv,
+		utils::{
+			currency::{register_currency, CurrencyInfo, CONST_DEFAULT_CUSTOM},
+			genesis,
+			genesis::Genesis,
+		},
+	};
+
+	pub struct LocalUSDC;
+	impl CurrencyInfo for LocalUSDC {
+		fn id(&self) -> CurrencyId {
+			CurrencyId::LocalAsset(LocalAssetId(1))
+		}
+
+		fn symbol(&self) -> &'static str {
+			"LocalUSDC"
+		}
+
+		fn custom(&self) -> CustomMetadata {
+			CustomMetadata {
+				pool_currency: true,
+				transferability: CrossChainTransferability::None,
+				..CONST_DEFAULT_CUSTOM
+			}
+		}
+	}
+
+	pub struct DomainUSDC;
+	impl CurrencyInfo for DomainUSDC {
+		fn id(&self) -> CurrencyId {
+			CurrencyId::ForeignAsset(100_0001)
+		}
+
+		fn symbol(&self) -> &'static str {
+			"DomainUSDC"
+		}
+
+		fn custom(&self) -> CustomMetadata {
+			CustomMetadata {
+				pool_currency: true,
+				transferability: CrossChainTransferability::LiquidityPools,
+				local_representation: Some(LocalAssetId(1)),
+				..Default::default()
+			}
+		}
+	}
+
+	#[test]
+	fn _test() {
+		local_to_variant::<centrifuge_runtime::Runtime>()
+	}
+
+	fn get_rate<T: Runtime>(key: (CurrencyId, CurrencyId)) -> Ratio {
+		let mut env = RuntimeEnv::<T>::default();
+
+		env.parachain_state_mut(|| {
+			assert_eq!(
+				orml_asset_registry::Metadata::<T>::get(LocalUSDC.id()),
+				None
+			);
+			assert_eq!(
+				orml_asset_registry::Metadata::<T>::get(DomainUSDC.id()),
+				None
+			);
```

Comment:
@wischli @lemunozm these assertions are triggered. Why on earth should this storage be populated already? I either am looking at the reason without seeing it, or something is really off here.

Reply:
This could be due to the migration being performed before initializing the tests? Our migration populates the state with those currencies IIRC

Reply:
Oh man. That could be. Haha. But then it is not working as intended. As the local representation is missing on asset 100_001. I will try to use different ids. 

Thanks @lemunozm!

Reply:
But we are using id `100_0001` instead of `100_001` here. Compiling right now with my cherry-pick for disabling migrations in std: https://github.com/centrifuge/centrifuge-chain/pull/1745/commits/0e53752f0d491afb5b9c03f6a35202f706a183a7